nomad/structs/structs.go

```diff
@@ -13282,3 +13285,9 @@ func NewRpcError(err error, code *int64) *RpcError {
 func (r *RpcError) Error() string {
 	return r.Message
 }
+
+type Action struct {
+	Name    string
+	Command *string
+	Args    []string
+}
```

Comment:
You'll need to implement `Copy` and `Equals`, as well as diff logic in `nomad/structs/diff.go`, in order to update jobs that have actions and to avoid potentially corrupting the state store. There's a [jobspec checklist](https://github.com/hashicorp/nomad/blob/main/contributing/checklist-jobspec.md) which can help you make sure you've got all the required bits.

I know this is a "scaffolding" PR though so whether this is blocking depends on how much scaffolding you'll need for the next chunk of work you'll do.

Reply:
Thanks, putting those together now. Should I be handling this in a new /nomad/structs/actions.go or keep them in /structs.go, in your opinion?

Reply:
It'd be nice to have them in `nomad/structs/actions.go`, just because that `structs.go` file is _huge_ :grinning: